### ISSUE TITLE
cuda softmax (and logsoftmax) are incorrect on very large tensors

### ISSUE THREAD

**Issue by rgreenblatt** [pytorch/pytorch#84144]
### 🐛 Describe the bug

Softmax (and logsoftmax) start returning incorrect values for tensors with `.nelement()` past `4 * 1024**3` (very large tensors).

```python
import torch

x = torch.randn(669, 128, 50259, device="cuda")
print("size", x.nelement() / (1024 ** 3))
torch.softmax(x, dim=-1)[-5:].sum(dim=-1) # returns either 0 or garbage for last several indexes (presumably the zeros also correspond to copied in garbage)
```

### Versions

```
PyTorch version: 1.13.0.dev20220826+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:06:46) [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-46-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.4.152
GPU models and configuration:
GPU 0: NVIDIA A100-SXM4-80GB
GPU 1: NVIDIA A100-SXM4-80GB
GPU 2: NVIDIA A100-SXM4-80GB
GPU 3: NVIDIA A100-SXM4-80GB
GPU 4: NVIDIA A100-SXM4-80GB
GPU 5: NVIDIA A100-SXM4-80GB
GPU 6: NVIDIA A100-SXM4-80GB
GPU 7: NVIDIA A100-SXM4-80GB

Nvidia driver version: 470.141.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] geotorch==0.3.0
[pip3] mypy==0.971
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.1
[pip3] torch==1.13.0.dev20220826+cu113
[pip3] torchaudio==0.13.0.dev20220826+cu113
[pip3] torchvision==0.14.0.dev20220826+cu113
[conda] cudatoolkit               11.7.0              hd8887f6_10    conda-forge
[conda] geotorch                  0.3.0                    pypi_0    pypi
[conda] numpy                     1.21.2                   pypi_0    pypi
[conda] torch                     1.13.0.dev20220826+cu113          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20220826+cu113          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220826+cu113          pypi_0    pypi
```

I also tried with latest stable instead of nightly. Both are incorrect in the same way.

**Comment by ngimel** [comment]
Thanks for reporting, the fix is in the linked PR, but may I ask what the usecase is? Even with half data type the input would need to be at least 4 GB, so even on a 40GB card you'd only be able to compute just a few layers of the neural network with inputs this big.
